src/AppInstallerCommonCore/Downloader.cpp

```diff
@@ -121,6 +127,138 @@ namespace AppInstaller::Utility
         return result;
     }
 
+    void DownloadPWAInstaller(

```

Comment:
Feels like this method should be broken up into smaller pieces.  Like one for building the json, one for opening the connection, one for reading the properties, one for fetching the chunks.